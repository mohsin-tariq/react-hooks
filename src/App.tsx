import { BrowserRouter, Switch, Route } from 'react-router-dom'
import Navbar from './components/Navbar'
import Buttons from './pages/Buttons'
import Dashboard from './pages/Dashboard'

import Home from './pages/Home'

const App = (): JSX.Element => {
  return (
    <BrowserRouter>
      <Navbar />
      <div className="container">
        <Switch>
          <Route path="/dashboard" component={Dashboard} />
          <Route path="/buttons" component={Buttons} />
          <Route path="/" component={Home} />
        </Switch>
      </div>
    </BrowserRouter>
  )
}

export default App
