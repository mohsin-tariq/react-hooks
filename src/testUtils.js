
/** eslint-disable **/
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import '@testing-library/jest-dom/extend-expect'

const initialState = {
  zone: {
    data: [
      {
        countryCode: 'AD',
        countryName: 'Andorra',
        zoneName: 'Europe/Andorra',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
      {
        countryCode: 'AL',
        countryName: 'Albania',
        zoneName: 'Europe/Albania',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
    ],
  },
}

const mockStore = configureMockStore()

function render(
  ui,
  { store = mockStore(initialState), ...renderOptions } = {}
) {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
