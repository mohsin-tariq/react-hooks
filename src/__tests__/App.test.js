import { render } from '../testUtils'

import App from '../App'

it('Renders the connected app with initialState', () => {
  const { asFragment } = render(<App />)
  expect(asFragment()).toMatchSnapshot()
})
