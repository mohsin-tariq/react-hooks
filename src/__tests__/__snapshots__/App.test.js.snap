// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the connected app with initialState 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      aria-current="page"
      class="navbar-brand active"
      href="/"
    >
      Home
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarNavAltMarkup"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-link"
          href="/dashboard"
        >
          Dashboard
        </a>
      </div>
    </div>
  </nav>
  <div
    class="container"
  >
    <h1>
      Time Zone Component
    </h1>
    <select>
      <option>
        Andorra, Europe/Andorra
      </option>
      <option>
        Albania, Europe/Albania
      </option>
    </select>
    <div
      class="card mt-5"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-text"
        >
          Country Code: AD
        </h5>
        <h5
          class="card-text"
        >
          Country Name: Andorra
        </h5>
        <h5
          class="card-text"
        >
          Zone Name: Europe/Andorra
        </h5>
        <p
          class="card-title"
        >
          Date Time: Tue Feb 23 2021 12:48:50 GMT+0000 (Greenwich Mean Time)
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
